crates/re_data_ui/src/data.rs

```diff
@@ -172,11 +174,37 @@ impl DataUi for re_types::datatypes::UVec4D {
         &self,
         _ctx: &ViewerContext<'_>,
         ui: &mut egui::Ui,
-        _ui_layout: UiLayout,
+        ui_layout: UiLayout,
         _query: &re_data_store::LatestAtQuery,
         _db: &re_entity_db::EntityDb,
     ) {
-        ui.label(self.to_string());
+        label_for_ui_layout(ui, ui_layout, self.to_string());
+    }
+}
+
+impl DataUi for Range1D {
+    fn data_ui(
+        &self,
+        _ctx: &ViewerContext<'_>,
+        ui: &mut Ui,
+        ui_layout: UiLayout,
+        _query: &LatestAtQuery,
+        _db: &EntityDb,
+    ) {
+        label_for_ui_layout(ui, ui_layout, self.to_string());
+    }
+}
+
+impl DataUi for Range2D {
+    fn data_ui(
+        &self,
+        _ctx: &ViewerContext<'_>,
+        ui: &mut Ui,
+        ui_layout: UiLayout,
+        _query: &LatestAtQuery,
+        _db: &EntityDb,
+    ) {
+        label_for_ui_layout(ui, ui_layout, self.to_string());
     }
 }
```

Comment:
so before we had data ui implement for those we'd just generate a label automatically. Shouldn't that label generation use `label_for_ui_layout` instead?

Reply:
Are you referring to `text_ui()` (in `component_ui_registry.rs`)? I believe several code path lead to that function.

Looking at it a bit more... It's essentially doing a more fancy version of `label_for_ui_layout`, so these are possibly duplicates. `text_ui` uses monospace font though, I'm not sure why. Seems like some further cleaning is in order around here.

Reply:
See update PR description. There is much work to be done here still.